nir/src/main/scala/scala/scalanative/nir/Vals.scala

```diff
@@ -28,6 +28,47 @@ sealed abstract class Val {
     case Val.String(_) => Rt.String
   }
 
+  private def countBytes(s: String): Int = {
+    import Character.isDigit
+
+    // TODO: should be a compilation error, how to report?
+    def malformed() = ???
```

Comment:
`throw new IllegalArgumentException("malforrmed C string: " + string)` is the way to crash here.